src/query/storages/hive/hive/src/hive_parquet_block_reader.rs

```diff
@@ -266,4 +291,8 @@ impl HiveBlockReader {
             .next_block(&self.projected_schema, &self.hive_partition_filler, &part)
             .map_err(|e| e.add_message(format!(" filename of hive part {}", part.filename)))
     }
+
+    pub fn support_blocking_api(&self) -> bool {
+        self.operator.metadata().can_blocking()
```

Comment:
seems we need another flag to control them.

can we have a setting named `hive_source_blocking_read_mode` ?

Reply:
Databend-query setting?